cli/azd/pkg/tools/azcli/ad.go

```diff
@@ -26,39 +29,108 @@ type AzureCredentials struct {
 	ResourceManagerEndpointUrl string `json:"resourceManagerEndpointUrl"`
 }
 
-func (cli *azCli) CreateOrUpdateServicePrincipal(
+// AdService provides actions on top of Azure Active Directory (AD)
+type AdService interface {
```

Comment:
Since we are moving `AdService` out of the azcli interface, should we move `ad.go` out of the `tools/azcli` folder and use `pkg/graphsdk` or `pkg/azsdk` ?

Reply:
Yes, I like that.  Lets take a holistic pass for the other similar services in this namespace in a separate/focused PR.